Dispatcher/Models/ArticleTableViewDelegate.swift

```diff
@@ -0,0 +1,14 @@
+//
+//  ArticleCellDelegate.swift
+//  Dispatcher
+//
+//  Created by Ofek Cohen on 20/04/2023.
+//
+
+import Foundation
+
+protocol ArticleTableViewDelegate {
+    func navigateToArticle(_ articleTitle: String)
+    func addArticleToFavorites(_ articleTitle: String)
```

Comment:
Names of methods inside delegate protocols usually (according to the convention) are given names like "actionButtonDidPress" or "backButtonTapped". This way it describes the action that happened, and the code that should be executed (e.g. performing a navigation) is done inside the implementation itself.